eng/common/scripts/git-branch-push.ps1

```diff
@@ -41,33 +42,40 @@ param(
 # would fail the first time git wrote command output.
 $ErrorActionPreference = "Continue"
 
-if (!(git remote | ? {$_ -eq $RemoteName}))
+if ((git remote) -contains $RemoteName)
 {
-    Write-Host "git remote add $RemoteName $GitUrl"
-    git remote add $RemoteName $GitUrl
-    if ($LASTEXITCODE -ne 0)
-    {
-        Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
-        exit $LASTEXITCODE
-    }
+  Write-Host "git remote get-url $RemoteName"
+  $remoteUrl = git remote get-url $RemoteName
+  if ($remoteUrl -ne $GitUrl)
+  {
+    Write-Error "Remote with name $RemoteName already exists with an incompatible url [$remoteUrl] which should be [$GitUrl]."
+    exit 1
+  }
 }
-
-Write-Host "git fetch $RemoteName"
-git fetch $RemoteName
-if ($LASTEXITCODE -ne 0)
+else 
 {
-    Write-Error "Unable to fetch remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
+  Write-Host "git remote add $RemoteName $GitUrl"
+  git remote add $RemoteName $GitUrl
+  if ($LASTEXITCODE -ne 0)
+  {
+    Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
     exit $LASTEXITCODE
+  }
+}
+# Checkout to $PRBranch, create new one if not exists.
+Write-Host "git checkout $PRBranchName."
```

Comment:
Just to ensure better diagnostics can you move this logging next to the command that actually runs and then make it match the command.